File: fiberplane-charts/src/Mondrian/generator/utils/getXAxisFromTimeRange.ts
import type { Axis, TimeRange } from "../../types";
import { getTimeFromTimestamp } from "./getTimeFromTimestamp";

/**
 * Returns the X axis to display results for the given time range.
 */
export function getXAxisFromTimeRange(timeRange: TimeRange): Axis {
  return {
    minValue: getTimeFromTimestamp(timeRange.from),
    maxValue: getTimeFromTimestamp(timeRange.to),
  };
}

File: mondrian-charts/src/utils/get_x_axis_from_time_range.rs
use super::get_time_from_timestamp;
use crate::types::{Axis, TimeRange};

/// Returns the X axis to display results for the given time range.
pub(crate) fn get_x_axis_from_time_range(time_range: &TimeRange) -> Axis {
    Axis {
        min_value: get_time_from_timestamp(time_range.from),
        max_value: get_time_from_timestamp(time_range.to),
        ..Default::default()
    }
}
